project_task_code/models/project_task.py

```diff
@@ -8,26 +8,17 @@ class ProjectTask(models.Model):
     _inherit = 'project.task'
 
     code = fields.Char(
-        string='Task Number', required=True, default="/", readonly=True)
+        string='Task Number',
```

Comment:
I'm afraid the `/` default is used by the logic.
So you wouldn't touch this part of the code.